doc/man3/flux_future_create.adoc

```diff
@@ -0,0 +1,110 @@
+flux_future_create(3)
+=====================
+:doctype: manpage
+
+
+NAME
+----
+flux_future_create, flux_future_destroy, flux_future_get, flux_future_check, flux_future_then, flux_future_set_timeout - synchronize an activity
+
+
+SYNOPSIS
+--------
+ #include <flux/core.h>
+
+ typedef struct flux_future_struct flux_future_t;
+
+ typedef void (*flux_continuation_f)(flux_future_t *f, void *arg);
+
+ flux_future_t *flux_future_create (flux_reactor_t *r);
+
+ void flux_future_destroy (flux_future_t *f);
+
+ int flux_future_set_timeout (flux_future_t *f, double timeout);
+
+ int flux_future_get (flux_future_t *f, void *result);
+
+ bool flux_future_check (flux_future_t *f);
+
+ int flux_future_then (flux_future_t *f, flux_continuation_f cb, void *arg);
+
+
+DESCRIPTION
+-----------
+
+A Flux future represents some activity that may be completed with reactor
+watchers and/or message handlers.  It is intended to be wrapped
+by other classes that perform specific activities, providing a common idiom
+for synchronizing on the result.  Authors of classes that wrap
+futures should refer to `flux_future_set_init(3)`.
+
+`flux_future_create()` creates a future and associates it with a
+Flux reactor.  A class that uses futures typically wraps this in
+a class-specific method returning a `flux_future_t`.  A future is
+destroyed with `flux_future_destroy()`.
+
+`flux_future_check()` returns true if a call to `flux_future_get()`
+will not block.  Otherwise, it returns false and `flux_future_get()`
+may or may not block.
+
+`flux_future_get()` returns the result of a future, or -1 with
+errno set.  It may block if the result or error is not yet available.
+A class that uses futures typically wraps this in a class-specific
+method that accepts a `flux_future_t`.
+
+`flux_future_then()` sets up a continuation callback on the reactor
```

Comment:
OK,  if the continuation is optional, and the reactor that is passed into flux_future_create() is only used conditionally when a continuation is used, shouldn't the reactor be passed as a parameter to flux_future_then() instead?